src/relations/async_replication.py

```diff
@@ -106,9 +106,6 @@ def __init__(self, charm):
         self.framework.observe(
             self.charm.on.create_replication_action, self._on_create_replication
         )
-        self.framework.observe(
-            self.charm.on.promote_to_primary_action, self._on_promote_to_primary
-        )
```

Comment:
Moved to the main charm code, since it's no longer used only for async promotion.